### ISSUE TITLE
More line ending issues

### ISSUE THREAD

**Issue by jgm** [jgm/cmark#73]
Modifying the test suite to use cr line endings instead of lf, we get two failures.  (With crlf endings, we only get the second of these.)

```
Example 56 (lines 955-970) Setext headers
Foo
Bar
---

Foo
Bar
===

--- expected HTML
+++ actual HTML
@@ -1,6 +1,4 @@
-<p>Foo
-Bar</p>
-<hr />
-<p>Foo
-Bar
-===</p>
+<h2>Foo
+Bar</h2>
+<h1>Foo
+Bar</h1>

Example 493 (lines 6924-6931) Links
[Foo
  bar]: /url

[Baz][Foo bar]

--- expected HTML
+++ actual HTML
@@ -1 +1 @@
-<p><a href="/url">Baz</a></p>
+<p>[Baz][Foo bar]</p>

596 passed, 2 failed, 0 errored, 0 skipped
```

**Comment by jgm** [comment]
First case above is due to the hardcoded `\n` at src/blocks.c line 799.

**Comment by jgm** [comment]
Fixed by 6c416f20273cd427af18e25c0911ef6cf0da96d0
